safetensors/src/tensor.rs

```diff
@@ -91,19 +97,36 @@ pub struct SafeTensors<'data> {
 
 impl<'data> SafeTensors<'data> {
     /// Given a byte-buffer representing the whole safetensor file
-    /// parses it and returns the Deserialized form (No Tensor allocation).
-    pub fn deserialize<'in_data>(buffer: &'in_data [u8]) -> Result<Self, SafeTensorError>
+    /// parses the header, and returns the size of the header + the parsed data.
+    pub fn read_metadata<'in_data>(
+        buffer: &'in_data [u8],
+    ) -> Result<(usize, Metadata), SafeTensorError>
     where
         'in_data: 'data,
     {
         let arr: [u8; 8] = [
             buffer[0], buffer[1], buffer[2], buffer[3], buffer[4], buffer[5], buffer[6], buffer[7],
         ];
-        let n = u64::from_le_bytes(arr) as usize;
+        let n: usize = u64::from_le_bytes(arr)
```

Comment:
do we still need a `u64` here ?

Reply:
Yes, `usize` is plateform dependant (mostly `u64` nowadays).

`usize` is the size of a pointer. Which we need for indexing the buffer.

The cases where this would be incorrect seems pretty niche now, but still.